sormas-backend/src/main/java/de/symeda/sormas/backend/action/ActionFacadeEjb.java

```diff
@@ -0,0 +1,199 @@
+		target.setActionStatus(source.getActionStatus());
+
+		target.setActionContext(source.getActionContext());
+		if (source.getActionContext() != null) {
+			switch (source.getActionContext()) {
+			case EVENT:
+				target.setEvent(eventService.getByReferenceDto(source.getEvent()));
+				break;
+			default:
+				throw new UnsupportedOperationException(source.getActionContext() + " is not implemented");
+			}
+		} else {
+			target.setEvent(null);
+		}
+
+		return target;
+	}
+
+	public ActionDto toDto(Action action) {
+
+		if (action == null) {
+			return null;
+		}
+
+		ActionDto target = new ActionDto();
+		Action source = action;
+
+		target.setCreationDate(source.getCreationDate());
+		target.setChangeDate(source.getChangeDate());
+		target.setUuid(source.getUuid());
+
+		target.setCreatorUser(UserFacadeEjb.toReferenceDto(source.getCreatorUser()));
+		target.setDescription(source.getDescription());
+		target.setReply(source.getReply());
+		target.setReplyingUser(UserFacadeEjb.toReferenceDto(source.getReplyingUser()));
+		target.setPriority(source.getPriority());
+		target.setDate(source.getDate());
+		target.setStatusChangeDate(source.getStatusChangeDate());
+		target.setActionContext(source.getActionContext());
+		target.setActionStatus(source.getActionStatus());
+		target.setEvent(EventFacadeEjb.toReferenceDto(source.getEvent()));
+
+		return target;
+	}
+
+	@Override
+	public ActionDto saveAction(ActionDto dto) {
+
+		Action ado = fromDto(dto);
+		actionService.ensurePersisted(ado);
+		return toDto(ado);
+	}
+
+	@Override
+	public ActionDto getByUuid(String uuid) {
+		return toDto(actionService.getByUuid(uuid));
+	}
+
+	@Override
+	public void deleteAction(ActionDto actionDto) {
+
+		if (!userService.hasRole(UserRole.ADMIN)) {
+			throw new UnsupportedOperationException("Only admins are allowed to delete entities.");
+		}
+
+		Action action = actionService.getByUuid(actionDto.getUuid());
+		actionService.delete(action);
+	}
+
+	@Override
+	public List<String> getAllActiveUuids() {
+
+		User user = userService.getCurrentUser();
+		if (user == null) {
+			return Collections.emptyList();
+		}
+
+		return actionService.getAllActiveUuids(user);
+	}
+
+	@Override
+	public List<ActionDto> getAllActiveActionsAfter(Date date) {
+
+		User user = userService.getCurrentUser();
+		if (user == null) {
+			return Collections.emptyList();
+		}
+
+		return actionService.getAllActiveActionsAfter(date, user).stream().map(c -> toDto(c)).collect(Collectors.toList());
+	}
+
+	@Override
+	public List<ActionDto> getByUuids(List<String> uuids) {
+		return actionService.getByUuids(uuids).stream().map(c -> toDto(c)).collect(Collectors.toList());
+	}
+
+	@Override
+	public List<ActionDto> getList(ActionCriteria actionCriteria, Integer first, Integer max) {
+		return actionService.getList(actionCriteria, first, max).stream().map(c -> toDto(c)).collect(Collectors.toList());
+	}
+
+	@Override
+	public List<ActionStatEntry> statsByEvent(ActionCriteria actionCriteria) {
```

Comment:
Since this takes an ActionCriteria, the method should rather be called getStatsByCriteria or (which I'd prefer) only getActionStats

Reply:
OK